models/rydberg_calcs.py

```diff
@@ -1,6 +1,80 @@
-from arc import Cesium as cs
+from arc import Cesium as Cs
+from arc import LevelPlot
 import numpy as np
 from scipy.interpolate import interp1d
+import networkx as nx
+
+
+class AtomState:
+    def __init__(self, n=7, l=1, j=1.5, mj=1.5, environmentTemperature=0.):
+        self.atom = Cs()
+        self.n = n
+        self.l = l
+        self.j = j
+        self.mj = mj
+        self.environmentTemperature = environmentTemperature
+
+        # calculation variables
+        self.nmin = 6 - 2
+        self.nmax = self.n + 5
+        self.lmin = 0
+        self.lmax = self.l + 5
+
+    def get_transition_matrix(self):
+        self.levels = LevelPlot(self.atom)
+        self.levels.makeLevels(self.nmin, self.nmax, self.lmin, self.lmax)
+        self.levels.makeTransitionMatrix(environmentTemperature=self.environmentTemperature,
+                                         printDecays=False)
+
+        return self.levels.transitionMatrix
+
+    def get_shortest_decay(self, source=None, target=None):
```

Comment:
This function would specifically benefit from a docstring explaining behavior. (And tests, at least of the core shortest-paths computation logic, but that's a bigger time investment that I'm not going to insist on.)